### PR TITLE
[Ready] Makes the admin-only high frequency blade !FUN! and adds it to the spellbook

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

Reworks the admin-only high frequency blade and unfucks the sprites
adds it to wizard spellbook
It now does 10 damage, attacks fast, can slash through almost everything with cool visual effects and has an unique damage multiplier mechanic! It calculates the distance between your two mouse clicks, so if you carefully attack the corners of something you may reach up to a 4.5 multiplier!
Video: https://streamable.com/wpnkm3
Damage multiplier video: https://streamable.com/26xyvi

## Why It's Good For The Game

fukig ASESOME

## Changelog

:cl: fikou & urumasi (with stackoverflows send by shaps)
add: the wizard now has a new flashy weapon.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
fikout pls stop teasing me with streamable clips its banned in russia for god knows what reason i beg you

File: code/game/objects/items/weaponry.dm [hunk 1]
```diff
@@ -785,81 +785,135 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 	lefthand_file = 'icons/mob/inhands/weapons/staves_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/weapons/staves_righthand.dmi'
 
-//HF blade
-/obj/item/vibro_weapon
-	icon_state = "hfrequency0"
-	base_icon_state = "hfrequency"
+/obj/item/melee/moonlight_greatsword
```
Comment (Fikou) [inline code/game/objects/items/weaponry.dm:788]: ignore these changes i just put the sword at the end of the file

**Comment by JohnFulpWillard** [comment]
nice icon diffs

**Comment by CRITAWAKETS** [comment]
with the low base damage and how augs are a flat damage reduction, simply using an HOS trenchcoat and augs will... 
https://i.kym-cdn.com/entries/icons/original/000/018/016/You_may_not_have_parents_but_at_least_you_got__4427f3dea2c6ba880479abe5d36c9b4f.jpg
Oh, right.

IM FUCKIN INVINCIBLE

**Comment by Wallemations** [comment]
https://user-images.githubusercontent.com/66052067/148303506-9e79426a-45f3-455c-bd8c-f4862d9a2316.mp4

**Comment by tralezab** [comment]
i allow it

**Comment by tralezab** [comment]
but not for free

**Review by Krysonism — APPROVED** [review]

File: code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -785,81 +785,162 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
+	var/matrix/target_matrix = matrix(target.transform)
+	var/scale_x = SIGN(target_matrix.a)*sqrt((target_matrix.a**2)+(target_matrix.d**2)) || -target_matrix.d
+	var/scale_y = SIGN(target_matrix.e)*sqrt((target_matrix.e**2)+(target_matrix.b**2)) || target_matrix.b
+	target_matrix = target_matrix.Scale(1/scale_x, 1/scale_y)
+	var/x_offset = x_slashed - world.icon_size/2
+	var/y_offset = y_slashed - world.icon_size/2
+	pixel_x = (target_matrix.a*x_offset + target_matrix.b*y_offset) * scale_x + target.pixel_x + target.transform.c
+	pixel_y = (-target_matrix.b*x_offset + target_matrix.a*y_offset) * scale_y + target.pixel_y + target.transform.f
+	transform = transform.Turn(rand(1, 360))
+	animate(src, duration*0.5, color = COLOR_BLUE, transform = transform.Scale(2), alpha = 255)
```
[the remark this answers is not in the document]
Comment (Fikou) [reply code/game/objects/items/weaponry.dm:946]: if this doesnt work im suing you

**Comment by Fikou** [comment]
waiting on some decomp stuff

File: code/game/objects/items/weaponry.dm [hunk 3]
```diff
@@ -785,81 +785,170 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
+	block_chance = 25
 	sharpness = SHARP_EDGED
-	attack_verb_continuous = list("cuts", "slices", "dices")
-	attack_verb_simple = list("cut", "slice", "dice")
 	w_class = WEIGHT_CLASS_BULKY
 	slot_flags = ITEM_SLOT_BACK
-	hitsound = 'sound/weapons/bladeslice.ogg'
-	var/wielded = FALSE // track wielded status on item
+	var/wielded = FALSE
+	var/previous_x
```
Comment (Kylerace) [inline code/game/objects/items/weaponry.dm:827]: doc this and below

File: code/game/objects/items/weaponry.dm [hunk 4]
```diff
@@ -785,81 +785,170 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
+	var/previous_y
+	var/atom/previous_target
```
Comment (Kylerace) [inline code/game/objects/items/weaponry.dm:829]: you dont seem to handle this being deleted

**Review by Kylerace — CHANGES_REQUESTED** [review]
show me a good time jack

**Pull request merged**